gio/src/unix_fd_list.rs

```diff
@@ -42,39 +42,41 @@ pub trait UnixFDListExtManual: IsA<UnixFDList> + Sized {
     }
 
     #[doc(alias = "g_unix_fd_list_get")]
-    fn get(&self, index_: i32) -> Result<RawFd, glib::Error> {
+    fn get(&self, index_: i32) -> Result<OwnedFd, glib::Error> {
         unsafe {
             let mut error = ptr::null_mut();
-            let ret = ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let raw_fd =
+                ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let fd = OwnedFd::from_raw_fd(raw_fd);
```

Comment:
This needs to be moved into the `error.is_null()` branch or otherwise you might create an `OwnedFd` out of `-1`

Reply:
Done.